src/consensus/consensus.rs

```diff
@@ -327,7 +327,7 @@ impl Consensus {
                 // println!("Connected to {connected_peers}/{total_peers} peers");
                 state
                     .shard_validator
-                    .sync_with_new_validator(&validator)
+                    .sync_against_validator(&validator)
```

Comment:
I think we should move this outside consensus.

Can we delete this for now? Or do we need it both in register and in ReceivedFullProposal?